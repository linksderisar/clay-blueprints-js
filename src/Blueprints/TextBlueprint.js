import Blueprint from './Blueprint';

export default class TextBlueprint extends Blueprint {
    constructor(content = '') {
        super('$text');
        this._content = content;
        this._bound = false;
    }

    get content() {
        return this._content;
    }

    set content(value) {
        this._content = value;
    }

    bind(bound = true) {
        this._bound = bound;
        return this;
    }

    get bound() {
        return this._bound;
    }

    set bound(value) {
        this._bound = value;
    }

    toObject() {
        const obj = super.toObject();

        if (this.bound) {
            obj[':value'] = this._content;
        } else {
            obj['value'] = this._content;
        }

        return obj;
    }
}
