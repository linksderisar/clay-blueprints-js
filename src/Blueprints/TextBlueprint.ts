import AbstractBlueprint from "./Abstracts/AbstractBlueprint";

export default class TextBlueprint extends AbstractBlueprint {
        type = '$text';
        content: string = '';
        bound = false;

    constructor(content = '') {
        super();
        this.content = content;
        this.bound = false;
    }

    public getContent(): string {
        return this.content;
    }

    public setContent(content: string): this {
        this.content = content;
        return this;
    }

    bind(bound = true): this {
        this.bound = bound;
        return this;
    }

    isBound() {
        return this.bound;
    }

    setBound(value) {
        this.bound = value;
    }

    toObject() {
        const obj = super.toObject();

        if (this.bound) {
            obj[':value'] = this.content;
        } else {
            obj['value'] = this.content;
        }

        return obj;
    }
}
