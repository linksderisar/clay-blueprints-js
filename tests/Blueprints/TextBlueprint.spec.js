import TextBlueprint from '../../src/Blueprints/TextBlueprint';

it('converts text to JSON', () => {
    const textBlueprint = new TextBlueprint('some text');
    expect(textBlueprint.toObject()).toMatchSnapshot({
        id: expect.any(String)
    });
});

it('converts bound text to JSON', () => {
    const textBlueprint = new TextBlueprint('some text');
    textBlueprint.bind();
    expect(textBlueprint.toObject()).toMatchSnapshot({
        id: expect.any(String)
    });
});
