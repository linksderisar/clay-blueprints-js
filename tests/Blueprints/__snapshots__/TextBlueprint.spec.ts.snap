// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts bound text to JSON 1`] = `
Object {
  ":value": "some text",
  "id": Any<String>,
  "type": "$text",
}
`;

exports[`converts text to JSON 1`] = `
Object {
  "id": Any<String>,
  "type": "$text",
  "value": "some text",
}
`;
